From 6c228e865b50645a9c68c8289e1e4100b0ac2dc6 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:49 -0400
Subject: [fs] btrfs: Return EXDEV for cross file system snapshot

Message-id: <1407179109-13714-163-git-send-email-sandeen@redhat.com>
Patchwork-id: 87016
O-Subject: [RHEL7.1 PATCH 162/422] btrfs: Return EXDEV for cross file system snapshot
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Kusanagi Kouichi <slash@ac.auone-net.jp>

commit 23ad5b17dce0f09af82c071b26acac35a0ab892b
Author: Kusanagi Kouichi <slash@ac.auone-net.jp>
Date:   Thu Jan 30 16:32:02 2014 +0900

    btrfs: Return EXDEV for cross file system snapshot

    EXDEV seems an appropriate error if an operation fails bacause it
    crosses file system boundaries.

    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Kusanagi Kouichi <slash@ac.auone-net.jp>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index a6d8efa..641f9da 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1573,7 +1573,7 @@ static noinline int btrfs_ioctl_snap_create_transid(struct file *file,
   if (src_inode->i_sb != file_inode(file)->i_sb) {
    btrfs_info(BTRFS_I(src_inode)->root->fs_info,
        "Snapshot src from another FS");
-   ret = -EINVAL;
+   ret = -EXDEV;
   } else if (!inode_owner_or_capable(src_inode)) {
    /*
     * Subvolume creation is not restricted, but snapshots
-- 
1.7.1